src/AnalyticsEventReporter.ts

```diff
@@ -34,10 +34,31 @@ export class AnalyticsEventReporter implements AnalyticsEventService {
   with(payload: EventPayload): AnalyticsEventService {
     const currentPayload =
       this.payload === undefined ? payload : merge(this.payload, payload);
+    if (this.config.debug) {
+      console.log('[DEBUG] AnalyticsConfig Object:\n', this.config);
+      console.log(
+        '[DEBUG] Payload following call to `with`:\n',
```

Comment:
Not sure how big of a difference it is, but probably should say `with()` instead of `with`. Might be more semantically obvious that `with` is referring to the function.